src/finch/codegen/numpy_buffer.py

```diff
@@ -70,8 +81,17 @@ def deserialize_from_c(self, c_buffer):
         Update this buffer based on how the C call modified the NumpyCBuffer structure.
         """
 
+    def serialize_to_numba(self):
+        return self.arr
```

Comment:
I'm concerned that if the program calls resize on the result of serialize_to_numba, that the new array won't be passed back into this numpy buffer type. It's unfortunate, but `np.resize` returns a new object, so the reference in this numpy buffer class will have gone stale.

Reply:
that's fine, but we need to update the reference to the outer numpy buffer. We might instead pass a reference to the numpy array, and update the reference when needed?

Reply:
So calling `np.resize(arr, size)` in the Numba backend should modify an instance of NumpyBuffer that this `arr` came from? Like `numpy_buffer_resize_callback`?

IIUC I'm not sure how this can't be achieved - We would need some kind of a callback to NumpyBuffer object but we can't pass custom classes to jited functions.

Reply:
can we pass NamedTuples to jited functions?

Reply:
@willow-ahrens The last commit is my proposition for supporting "in-place" resize for Numba, but it also impacts C backend.

Let's wrap the NumPy array in the `NumpyBuffer` in a list and keep it as `self.arr_ref = [arr]`. This way in Numba I access it with `arr_ref[0]` and can update the reference to it.
For C backend it means that `arr_ref` is in the `NumpyCBuffer` but nothing else much.

WDYT? Is it too much of a hacky approach or still reasonable?